article/app/services/dotcomponents/OnwardsPicker.scala

```diff
@@ -4,11 +4,13 @@ import agents.CuratedContentAgent
 import common.Edition
 import model.dotcomrendering.OnwardCollectionResponse
 import model.{ArticlePage, ContentFormat}
+import play.api.mvc.RequestHeader
+import views.support.Commercial
 
 class OnwardsPicker(curatedContentAgent: CuratedContentAgent) {
-  def forArticle(article: ArticlePage, edition: Edition): Seq[OnwardCollectionResponse] = {
+  def forArticle(article: ArticlePage, edition: Edition)(implicit request: RequestHeader): Seq[OnwardCollectionResponse] = {
```

Comment:
We can add `isAddFree` as a boolean argument instead of passing the request as an implicit